### PR TITLE
Catch uglify error

### PR DESCRIPTION
**Pull request by tomaskikutis**
There are packages e.g. qs version 6.0.1 which crash the application because of uncaught uglify error.

### CODE REVIEW

**Comment by bendrucker** [comment]
Thanks for submitting—I'll need unit tests and correct code formatting to consider this.

**Comment by tomaskikutis** [comment]
I've fixed formatting and added a test case.

File: index.js [hunk 1]
```diff
@@ -15,8 +15,14 @@ module.exports = function browserifySize (name, options, callback) {
     if (err) return callback(err)
     bundle(dir, function (err, code) {
       if (err) return callback(err)
-      code = minify(code)
-      gzipSize(code, callback)
+      minify(code, function (error, minifiedCode) {
```
Comment (bendrucker) [inline index.js:18]: - error -> err
- err === null -> !err
- return early instead of using an if/else
- function (err, code)

File: index.js [hunk 2]
```diff
@@ -25,6 +31,11 @@ function bundle (dir, callback) {
   browserify(dir).bundle(callback)
 }
 
-function minify (code) {
-  return uglify.minify(code.toString(), {fromString: true}).code
+function minify (code, callback) {
+  try {
+    var minifiedCode = uglify.minify(code.toString(), {fromString: true}).code
```
Comment (bendrucker) [inline index.js:36]: code =

File: test.js [hunk 3]
```diff
@@ -22,3 +22,10 @@ test('remote', function (t) {
     t.ok(bytes < 500)
   })
 })
+
+test('uglify error should not crash the application', function (t) {
+  t.doesNotThrow(function () {
+    size('qs', {version: '6.0.1'}, function () {})
```
Comment (bendrucker) [inline test.js:28]: you need to manage the async operation in this test. you can't just end it immediately.

**Comment by bendrucker** [comment]
More requests added inline

File: index.js (hunk 2 above)
Comment (tomaskikutis) [inline index.js:36]: You shouldn't shadow variables, it can lead to nasty and hard to debug errors.

**Comment by tomaskikutis** [comment]
I think I've done enough even with the initial PR - fixed a real life problem. If you want to count semicolons I will leave it up to you.

**Pull request closed without merging**